storage/database.go

```
@@ -264,20 +257,38 @@ func (d *db) IsBootstrapped() bool {
 	return d.bsm.IsBootstrapped()
 }
 
-func (d *db) getOwnedShards() []databaseShard {
+func (d *db) TruncateNamespace(namespace string) error {
+	n, err := d.readableNamespace(namespace)
+	if err != nil {
+		return err
+	}
+	n.Truncate()
```

Truncate should probably return an error yeah?